src/utils/adapter-utils.ts

```diff
@@ -9,4 +10,40 @@ export function getAutobindWrapperAttributes(element: Element) {
     'original-source': element.tagName.toLowerCase()
   };
   return _.omitBy(attributes, _.isEmpty) as AutobindWrapperAttributes;
+}
+
+
+export function getEmbedCheckDataAsEmbeddableString(checkInformation: CheckInformationKeyValuePair[], inputFormat: string) {
+  let embeddableString = "";
+  const ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME = "acrolinxCheckData";
+
+  if (checkInformation) {
+    if (inputFormat === "HTML") {
+      embeddableString = embeddableString.concat("<meta " + "name=\"" + ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME + "\" ");
+      checkInformation.forEach(element => {
+        embeddableString = embeddableString.concat(element.key).concat("=").concat("\"").concat(element.value).concat("\" ");
+        embeddableString = embeddableString.concat("/>");
+        return embeddableString;
+      });
+    }
+    else if (inputFormat === "XML") {
+      let embeddableString = "";
+      embeddableString = embeddableString.concat("<?" + ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME + " ");
+      checkInformation.forEach(element => {
+        embeddableString = embeddableString.concat(element.key).concat("=").concat("\"").concat(element.value).concat("\" ");
+        embeddableString = embeddableString.concat("?>");
+        return embeddableString;
+      });
+    }
+    else if (inputFormat === "MARKDOWN") {
+      let embeddableString = "";
+      embeddableString = embeddableString.concat("<!-- " + ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME + " ");
+      checkInformation.forEach(element => {
+        embeddableString = embeddableString.concat(element.key).concat("=").concat("\"").concat(element.value).concat("\" ");
+        embeddableString = embeddableString.concat("-->");
+        return embeddableString;
+      });
+    }
+  };
+  return embeddableString;
 }
```

Comment:
Like Ralf has already written by email, I think we should implement a different approach.

However I would still like to review your getEmbedCheckDataAsEmbeddableString function,
just in case you want to implement a similar function in the future.


*Errors*

* It returns an empty string for all inputFormats other than HTML.
* It does not work if checkInformation.length>1 because than it generates
  broken tags like `<meta name="acrolinxCheckData" key="value" />key2="value2" />`
* The values are not escaped, which means that values like `it can"happen` will create
  invalid tags like `<meta name="acrolinxCheckData" key="it can"happen" />`


Useless code:
* `return embeddableString;` at the end of the forEach blocks does nothing

Confusing:
* `let embeddableString` inside of the elseif blocks shadows the outer embeddableString variable
* In my experience, it is very unusual to use `concat` to append strings in the (chained) way it's done here,
  because it creates a lot of unnecessary code
  Better alternatives: template strings, `+` operator, array.join()
* I think an switch statement would be more appropriate instead of the multiple elseif blocks
* An `if` statement should not have a semicolon at the end

Here an example how i would implement the function:

```typescript
export function getEmbedCheckDataAsEmbeddableString2(checkInformation: CheckInformationKeyValuePair[], inputFormat: string) {
  if (!checkInformation) {
    return '';
  }
  
  const ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME = "acrolinxCheckData";
  const attributes: string = checkInformation.map(({key, value}) => `${key}="${_.escape(value)}"`).join(' ');

  switch (inputFormat) {
    case 'HTML':
      return `<meta name="${ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME}" ${attributes} />`;
    case 'XML':
      return `<? ${ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME} ${attributes} ?>`;
    case 'MARKDOWN':
      return `<!-- name="${ACROLINX_PROCESSING_INSTRUCTION_TAG_NAME}" ${attributes} -->`;
    default:
      return '';
  }
}

```

Reply:
Thanks @acromarco  for reviewing. I will take care of all the suggestions and best practices while implementing.

As @acrork had suggested I have added a callback function in config for passing check data to the plugin.